### ISSUE TITLE
Loading flag after redirect to App side and before handleRedirectPromise()

### ISSUE THREAD

**Issue by Tzvetelin88** [Azure-Samples/ms-identity-javascript-react-spa-dotnetcore-webapi-obo#34]
Please follow the issue template below. Failure to do so will result in a delay in answering your question.

## Library

- [ ] `msal@1.x.x` or `@azure/msal@1.x.x`
- [X] `@azure/msal-browser@2.x.x`

## Description

I'm using React. Is there a flag which is set on MSAL after redirect back to Application side and before handleRedirectPromise() ?

handleRedirectPromise() need few seconds to get the response users can still see Login page on App side.
I checked a lot of examples with React and MSAL 2.x and all of them have this issue :)

So basically the steps:

User clicks Login on App side.
User is redirected to MS page for Pass/MFA
User is redirected to App side.
User is seeing Login page again
After few seconds needed for handleRedirectPromise() to get response.
User is authenticated and redirected to Welcome page.

I read few articles and some people recommend to use new empty page before send user to Login page. But as I'm using React is there a cleaver way to achieve that?

**Comment by Tzvetelin88** [comment]
Actually Local Storage is preserved after refresh, and I used Persist Reducer to set Flag in the store. 
When user clicks SignIn, flag is set to true and then redirects to MS page. 

On return back, as we have flag true I show loading and after  handleRedirectPromise return a promise I redirects user to Welcome page.
